Public/Src/Pips/Dll/Operations/Process.cs

```diff
@@ -765,6 +793,119 @@ public bool TryComputePipUniqueOutputHash(PathTable pathTable, out long pipUniqu
             return true;
         }
 
+        /// <summary>
+        /// Compute the intersection of the pip's dependencies and global affected SourceChangeAffectedOutputs of the build
+        /// </summary>
+        public ReadOnlyArray<string> GetChangeAffectedInputNames(PathTable pathTable)
+        {
+            var changeAffectedInputs = SourceChangeAffectedOutputFiles.Intersect(Dependencies.Select(d => d.Path)).Select(i => i.GetName(pathTable).ToString(pathTable.StringTable)).ToHashSet();

```

Comment:
>. [](start = 114, length = 1)

nit: Small formatting suggestion since there are so many parenthesis:

I usually find it easier to read code when deep exressions like this are left-aligh the larger dots and then indent i.e. something like
```
SourceChangeAffectedOutputFiles
    .Intersect(Dependencies.Select(d => d.Path))
    .Select(i => i.GetName(pathTable).ToStringpathTable.StringTable))
    .ToHashSet();
```